x/incentives/keeper/distribute.go

```diff
@@ -218,67 +222,63 @@ func (k Keeper) doDistributionSends(ctx sdk.Context, distrs *distributionInfo) e
-// func (k Keeper) distributeSyntheticInternal(
-// 	ctx sdk.Context, gauge types.Gauge, locks []lockuptypes.PeriodLock, distrInfo *distributionInfo) (sdk.Coins, error) {
-// 	totalDistrCoins := sdk.NewCoins()
-// 	denom := gauge.DistributeTo.Denom
-// 	suffix := lockuptypes.SyntheticSuffix(denom)
-
-// 	synthLocks := []lockuptypes.SyntheticLock{}
-// 	for _, lock := range locks {
-// 		synthLock, err := k.lk.GetSyntheticLockup(ctx, lock.ID, suffix)
-// 		if err != nil {
-// 			k.Logger(ctx).Error(err.Error())
-// 			continue
-// 		}
-// 		synthLocks = append(synthLocks, *synthLock)
-// 	}
-
-// 	lockSum := lockuptypes.SumSyntheticLocksByDenom(synthLocks, denom)
-
-// 	if lockSum.IsZero() {
-// 		return nil, nil
-// 	}
-
-// 	remainCoins := gauge.Coins.Sub(gauge.DistributedCoins)
-// 	remainEpochs := uint64(1)
-// 	if !gauge.IsPerpetual { // set remain epochs when it's not perpetual gauge
-// 		remainEpochs = gauge.NumEpochsPaidOver - gauge.FilledEpochs
-// 	}
-
-// 	for _, synthLock := range synthLocks {
-// 		distrCoins := sdk.Coins{}
-// 		for _, coin := range remainCoins {
-// 			// distribution amount = gauge_size * denom_lock_amount / (total_denom_lock_amount * remain_epochs)
-// 			denomLockAmt := synthLock.Coins.AmountOfNoDenomValidation(denom)
-// 			amt := coin.Amount.Mul(denomLockAmt).Quo(lockSum.Mul(sdk.NewInt(int64(remainEpochs))))
-// 			if amt.IsPositive() {
-// 				newlyDistributedCoin := sdk.Coin{Denom: coin.Denom, Amount: amt}
-// 				distrCoins = distrCoins.Add(newlyDistributedCoin)
-// 			}
-// 		}
-// 		distrCoins = distrCoins.Sort()
-// 		if distrCoins.Empty() {
-// 			continue
-// 		}
-// 		// Update the amount for that address
-// 		err := distrInfo.addLockRewards(synthLock.Owner, distrCoins)
-// 		if err != nil {
-// 			return nil, err
-// 		}
-
-// 		totalDistrCoins = totalDistrCoins.Add(distrCoins...)
-// 	}
-
-// 	// increase filled epochs after distribution
-// 	gauge.FilledEpochs += 1
-// 	gauge.DistributedCoins = gauge.DistributedCoins.Add(totalDistrCoins...)
-// 	if err := k.setGauge(ctx, &gauge); err != nil {
-// 		return nil, err
-// 	}
-
-// 	return totalDistrCoins, nil
-// }
+// TODO: Make this code have way more re-use with distribute internal (post-v7)
+func (k Keeper) distributeSyntheticInternal(
+	ctx sdk.Context, gauge types.Gauge, locks []lockuptypes.PeriodLock, distrInfo *distributionInfo) (sdk.Coins, error) {
+	totalDistrCoins := sdk.NewCoins()
+	denom := gauge.DistributeTo.Denom
+	suffix := lockuptypes.SyntheticSuffix(denom)
+
+	synthLocks := []lockuptypes.SyntheticLock{}
+	for _, lock := range locks {
+		synthLock, err := k.lk.GetSyntheticLockup(ctx, lock.ID, suffix)
+		if err != nil {
+			k.Logger(ctx).Error(err.Error())
+			continue
+		}
+		synthLocks = append(synthLocks, *synthLock)
+	}
+
+	lockSum := lockuptypes.SumSyntheticLocksByDenom(synthLocks, denom)
+
+	if lockSum.IsZero() {
+		return nil, nil
+	}
+
+	remainCoins := gauge.Coins.Sub(gauge.DistributedCoins)
```

Comment:
Note to reviewer: This function from here on is ~entirely copy-paste of the normal distribute code. I tried making a common function via interface, ran into an issue so stopped looking into it. But its probably not that insightful to review in detail.

Didn't spend too much time working through the interface issue, since this changes anyway with the complete F1 code.